### ISSUE TITLE
Searches no longer trigger search events, or register in statistics

### ISSUE THREAD

**Issue by MarieVerdonck** [DSpace/dspace-angular#1811]
**Describe the bug**
A search, on the regular search page or from the search box in header, no longer trigger a `POST /server/api/statistics/searchevents` request, so the search is not logged in the statistics.

**To Reproduce**
1. Do a few searches on /search page, some of which unique
2. Verify in network tab there is not request done like `POST /server/api/statistics/searchevents` body: `{query: "test", page: {totalElements: 205, totalPages: 21, number: 1},…}`

**Expected behavior**
A search on the regular search page should trigger a search event, and as such get logged as a search statistic

**Proposed solution**

/searchevents request is done by [StatisticsService#trackSearchEvent](https://github.com/DSpace/dspace-angular/blob/main/src/app/statistics/statistics.service.ts#L49) which is controlled by library angulartics2.
The tracking is now done separately from the actual search component in search-tracker.component.ts [here](https://github.com/DSpace/dspace-angular/blob/main/src/app/search-page/search-tracker.component.ts#L64-L81), but this doesn’t always trigger any more. Move logic to `SearchComponent`

**Comment by MarieVerdonck** [comment]
This will take an estimated **8 hours** to refactor and test

**Comment by tdonohue** [comment]
@MarieVerdonck : I agree this seems like a serious bug.  I'll add the estimate & assign back to you for you/your team to work on for 7.4.  Thanks!
